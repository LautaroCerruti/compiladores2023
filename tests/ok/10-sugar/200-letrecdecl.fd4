let x : Nat = let rec g(x1:Nat)(x2:Nat):Nat = ifz (x1-x2) then 0 else g (x1) (x2+1) in g 5 1

let z : Nat = let rec h(x:Nat):Nat = ifz x then 0 else h (x-1) in h 5

let rec f(x1:Nat)(x2:Nat):Nat = ifz (x1-x2) then 0 else print "dif " (f (x1) (x2+1))

let y : Nat = f 5 1