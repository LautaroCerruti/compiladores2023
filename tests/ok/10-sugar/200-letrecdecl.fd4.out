dif 0
dif 0
dif 0
dif 0
